core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -321,6 +321,16 @@ package object config {
       .intConf
       .createWithDefault(3)
 
+  private[spark] val REDUCER_MAX_BLOCKS_IN_FLIGHT_PER_ADDRESS =
+    ConfigBuilder("spark.reducer.maxBlocksInFlightPerAddress")
+      .doc("This configuration limits the number of remote blocks being fetched from a given " +
+        " host port at any given point. When external shuffle is enabled and a large number of " +
```

Comment:
In this case it would take down either the NM or the executor serving the map output tasks based on the shuffle mode. We emphasize the external shuffle case as crashing the NM is more severe than loosing an executor. I am open to re-wording it so that it is easier to understand.